app/src/androidTest/java/com/android/sample/ui/listActivities/ListActivitiesTest.kt

```diff
@@ -250,31 +255,41 @@ class OverviewScreenTest {
   }
 
   @Test
-  fun filteringWorks() {
+  fun filteringCategoryWorks() {
     userProfileViewModel = mock(ProfileViewModel::class.java)
     `when`(userProfileViewModel.userState).thenReturn(MutableStateFlow(testUser))
     composeTestRule.setContent {
       ListActivitiesScreen(
           listActivitiesViewModel, navigationActions, userProfileViewModel, locationViewModel)
     }
-    val activity1 = activity.copy(title = "cooking", type = ActivityType.INDIVIDUAL)
-    val activity2 = activity.copy(title = "dance", type = ActivityType.SOLO)
-    val activity3 = activity.copy(title = "football", type = ActivityType.INDIVIDUAL)
+    val activity1 = activity.copy(title = "Italian cooking", category = Category.CULTURE)
+    val activity2 = activity.copy(title = "dance", category = Category.ENTERTAINMENT)
+    val activity3 = activity.copy(title = "football", category = Category.SPORT)
+    val activity4 = activity.copy(title = "networking", category = Category.SKILLS)
 
     `when`(activitiesRepository.getActivities(any(), any())).then {
-      it.getArgument<(List<Activity>) -> Unit>(0)(listOf(activity, activity1, activity2, activity3))
+      it.getArgument<(List<Activity>) -> Unit>(0)(
+          listOf(activity, activity1, activity2, activity3, activity4))
     }
 
     listActivitiesViewModel.getActivities()
-    composeTestRule.onNodeWithTag("segmentedButtonINDIVIDUAL").performClick()
-    composeTestRule.onNodeWithText("cooking").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("segmentedButtonSPORT").performClick()
     composeTestRule.onNodeWithText("football").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("segmentedButtonSPORT").performClick()
 
-    composeTestRule.onNodeWithTag("segmentedButtonSOLO").performClick()
+    composeTestRule.onNodeWithTag("segmentedButtonENTERTAINMENT").performClick()
     composeTestRule.onNodeWithText("dance").assertIsDisplayed()
+    composeTestRule.onNodeWithText("football").assertIsNotDisplayed()
+    composeTestRule.onNodeWithText("Italian cooking").assertIsNotDisplayed()
+    composeTestRule.onNodeWithText("networking").assertIsNotDisplayed()
+    composeTestRule.onNodeWithTag("segmentedButtonENTERTAINMENT").performClick()
 
-    composeTestRule.onNodeWithTag("segmentedButtonPRO").performClick()
-    composeTestRule.onNodeWithText("Mountain Biking").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("segmentedButtonCULTURE").performClick()
+    composeTestRule.onNodeWithText("Italian cooking").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("segmentedButtonCULTURE").performClick()
+
+    composeTestRule.onNodeWithTag("segmentedButtonSKILLS").performClick()
+    composeTestRule.onNodeWithText("networking").assertIsDisplayed()
```

Comment:
Great mocking and instantiating involved! Clever way to check for the categories. However, for the purposes of test clarity, you could have chosen names than are closer to the categories. Such as directly writing "Sport Activity", "Entertainment Activity"...